src/main/java/gui/view/MainPageController.java

```diff
@@ -0,0 +1,159 @@
+package gui.view;
+
+import com.jfoenix.controls.JFXDrawer;
+import com.jfoenix.controls.JFXListView;
+import datagateway.event.EventReader;
+import datagateway.task.TaskReader;
+import gui.utility.NavigationHelper;
+import gui.viewmodel.MainPageViewModel;
+import gui.viewmodel.ViewModel;
+import javafx.fxml.FXML;
+import javafx.fxml.Initializable;
+import javafx.scene.control.Label;
+import javafx.scene.layout.HBox;
+import javafx.scene.text.Font;
+
+import java.net.URL;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.FormatStyle;
+import java.util.*;
+
+public class MainPageController implements Initializable, ViewModelBindingController {
+
+    private MainPageViewModel viewModel;
+    final double labelFontSize = 15;
+
+    @FXML
+    private JFXDrawer collapsedNavPanel;
+
+    @FXML
+    private JFXDrawer extendedNavPanel;
+
+    @FXML
+    private JFXListView<HBox> eventListView;
+
+    @FXML
+    private JFXListView<HBox> taskListView;
+
+
+    @Override
+    public void initialize(URL location, ResourceBundle resources) {
+        NavigationHelper.initializeNavPanel(extendedNavPanel, collapsedNavPanel);
+    }
+
+    @Override
+    public void init(ViewModel viewModel) {
+        this.viewModel = (MainPageViewModel) viewModel;
+
+        this.viewModel.addObserver(this::acceptReaders);
+
+        updateListView(taskListView, formatTaskReaders(this.viewModel.getRelevantTasks()));
+        updateEventListView(eventListView, formatEventReaders(this.viewModel.getRelevantEvents()));
+
+    }
+
+    /**
+     * Note: removes visibility of all previously rendered labels in the list view before flushing it out.
+     * @param listView the list view to update
+     * @param map a mapping of name and time for today's relevant items
+     */
+    private void updateListView(JFXListView<HBox> listView, Map<String, String> map) {
```

Comment:
- method name `updateTaskListView`?

- Since this is for task list, I think the parameter `JFXListView<HBox> listView` is not necessary